index.d.ts

```diff
@@ -1,14 +1,16 @@
-declare module 'before-after-hook' {
-  export type HookType = {
-    new (): any
-    (name: string, data: any, method: (...any) => any): Promise<any>
-    before (name: string, method: (...any) => any): Promise<any>
-    error (name: string, method: (...any) => any): Promise<any>
-    after (name: string, method: (...any) => any): Promise<any>
-    wrap (name: string, method: (...any) => any): Promise<any>
-    remove (name: string, beforeHookMethod: (...any) => any): Promise<any>
-  }
+interface HookInstance {
+  (name: string | string[], method: (options: any) => any): Promise<any>
+  (name: string | string[], options: any, method: (options: any) => any): Promise<any>
+  before (name: string, method: (options: any) => any): HookInstance
```

Comment:
like this?

```suggestion
  /**
   * Add before hook for given name. Returns hook instance for chaining.
   */
  before (name: string, method: (options: any) => any): HookInstance
```

Reply:
Yep! Results in this:

<img width="543" alt="image" src="https://user-images.githubusercontent.com/10660468/50541556-71837d80-0b76-11e9-87a4-5d42065c415e.png">

Note the comment at the bottom of that annotation!

Reply:
Awesome! I’m busy elsewhere right now, do you want to add these?

Reply:
Done 👍 as of 4b071d2